packages/modules/workflow-engine-redis/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -438,19 +466,12 @@ export class RedisDistributedTransactionStorage
         schedulerOptions,
       },
       {
-        repeat: {
-          pattern: schedulerOptions.cron,
-          limit: schedulerOptions.numberOfExecutions,
-          key: `${JobType.SCHEDULE}_${jobId}`,
-        },
+        repeat: repeatOptions,
         removeOnComplete: {
           age: 86400,
           count: 1000,
         },
-        removeOnFail: {
-          age: 604800,
-          count: 5000,
-        },
+        removeOnFail: false,
```

Comment:
I think you typically don't want to store failures indefinitely, you need them long-enough to debug them. Instead of removing it completely, we can increase the age if you want, but I think we should still keep it.